openwisp_radius/api/views.py

```diff
@@ -1,8 +1,49 @@
+from django.contrib.auth.models import AnonymousUser
+from django.core.cache import cache
+from django_freeradius.api.views import AccountingView as BaseAccountingView
+from django_freeradius.api.views import AuthorizeView as BaseAuthorizeView
 from django_freeradius.api.views import BatchView as BaseBatchView
-from django_freeradius.api.views import accounting, authorize, postauth  # noqa
+from django_freeradius.api.views import PostAuthView as BasePostAuthView
+from django_freeradius.api.views import TokenAuthentication as BaseTokenAuthentication
+from rest_framework.exceptions import AuthenticationFailed
 
 from openwisp_users.models import Organization
 
+from ..models import OrganizationRadiusSettings
+
+
+class TokenAuthentication(BaseTokenAuthentication):
+    def authenticate(self, request):
+        id = None
+        token = None
+        if request.META.get('HTTP_AUTHORIZATION'):
+            headers = request.META.get('HTTP_AUTHORIZATION').split(',')
+            for header in headers:
+                try:
+                    id = header.split(' ')[1]
+                    token = header.split(' ')[2]
+                except IndexError:
+                    raise AuthenticationFailed('Token authentication failed')
+        elif request.GET.get('token') and request.GET.get('id'):
+            id = request.GET.get('id')
+            token = request.GET.get('token')
+
+        if id and token:
+            if cache.get(id):
+                if cache.get(id).token != token:
+                    raise AuthenticationFailed('Token authentication failed')
+                return (AnonymousUser(), None)
+            else:
+                try:
+                    instance = OrganizationRadiusSettings.objects.get(organization=id,
+                                                                      token=token)
+                    cache.set(str(instance.pk), instance)
+                    return (AnonymousUser(), None)
+                except OrganizationRadiusSettings.DoesNotExist:
+                    raise AuthenticationFailed('Token authentication failed')
+
+        raise AuthenticationFailed('Token authentication failed')
```

Comment:
Here I think you can do:

```python
    return (AnonymousUser(), None)
raise AuthenticationFailed('Token authentication failed')
```
This way `return (AnonymousUser(), None)` is written only once and the code will get there only if `AuthenticationFailed` was not raised so it should work.
If `uuid` and `token` are `None` the `if` block won't executed and the execution should jump to the last raise, right?